### PR TITLE
Release process

### PR DESCRIPTION
**Pull request by Philmod**
Use a GCB build to release a new version of kagglehub both on pypi and in this repo as a GitHub release.

I ran it for `0.1.5`: 
- pypi: https://pypi.org/project/kagglehub/0.1.5/#history
- GitHub release: https://github.com/Kaggle/kagglehub/releases/tag/v0.1.5

http://b/314006079

### CODE REVIEW

**Comment by Philmod** [comment]
We could add more automation by doing the `hatch version $_NEW_VERSION` in the build, but that means giving the build access to write to GitHub main branch, but a human would still have to deal with the CHANGELOG so not sure if it's worth it.

File: tools/release/cloudbuild.yaml [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  - test
+
+- name: gcr.io/$PROJECT_ID/hatch:${_PYTHON_VERSION}
+  id: lint
+  dir: $_GITHUB_REPOSITORY
+  args:
+  - run
+  - lint:all
+
+# Delete old releases if exist.
```
Comment (rosbo) [inline tools/release/cloudbuild.yaml:64]: Is this necessary given the `gcr.io/$PROJECT_ID/hatch` image probably doesn't have any pre-baked?

File: tools/release/cloudbuild.yaml [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  - |
+    twine upload dist/* -r pypi
+  volumes:
+  - name: 'root'
+    path: /root
+
+substitutions:
+  _GITHUB_REPOSITORY: kagglehub
+  _PYTHON_VERSION: 3.9.18
+  _NEW_VERSION: 0.0.1
```
Comment (rosbo) [inline tools/release/cloudbuild.yaml:118]: Can we set `_NEW_VERSION` to a value that would throw an error if the user doesn't specify it in the `gcloud submit` command?

**Review by rosbo — APPROVED** [review]
Thanks so much for setting this up!

**Comment by rosbo** [comment]
> We could add more automation by doing the `hatch version $_NEW_VERSION` in the build, but that means giving the build access to write to GitHub main branch, but a human would still have to deal with the CHANGELOG so not sure if it's worth it.

Agree. Given that we manually update the CHANGELOG for now, let's keep it the way you have implemented it.

File: tools/release/cloudbuild.yaml (hunk 2 above)
Comment (Philmod) [reply tools/release/cloudbuild.yaml]: Added some validation

**Pull request merged**